modules/tinymce/src/core/main/ts/fmt/CaretFormat.ts

```diff
@@ -378,8 +353,9 @@ const isFormatElement = (editor: Editor, element: SugarElement<Node>): boolean =
   return Obj.has(inlineElements, SugarNode.name(element)) && !isCaretNode(element.dom) && !NodeType.isBogus(element.dom);
 };
 
-const isEmptyCaretFormatElement = (element: SugarElement<Node>): boolean => {
-  return isCaretNode(element.dom) && isCaretContainerEmpty(element.dom);
+const isFormatCaret = (editor: Editor, element: SugarElement<Node>): boolean => {
```

Comment:
Is this function just to match when the current node contains a caret and is a text node? This name is a little confusing. You could also return the `isCaretNode` match first, then call and match the inline elements second to run faster when not in a caret node.

Reply:
nope its the reverse, checking if the element is not a bogus or caret, and is a format element such as`b/i/u/span`

Reply:
ah i see what u mean, true that should speed thing up a bit